lib/modules/datasource/docker/index.ts

```diff
@@ -766,7 +776,10 @@ export class DockerDatasource extends Datasource {
       if (err instanceof ExternalHostError) {
         throw err;
       }
-      if (err.statusCode === 404 && !dockerRepository.includes('/')) {
+      if (
+        (err.statusCode === 404 || err.message === PAGE_NOT_FOUND_ERROR) &&
```

Comment:
If we already had a check for err.statusCode === 404, what is the reason for needing to convert it to PAGE_NOT_FOUND_ERROR at a lower level? i.e. what change of logic does this cause compared to before?

Reply:
Best to explain this in the code itself with comments too, because it's not intuitive as-is